pins/cache.py

```diff
@@ -57,12 +57,62 @@ def prefix_cache(fs, board_base_path):
     return f"{proto_name}_{base_hash}"
 
 
+class HashMapper:
```

Comment:
rather than having different classes for each cache type, we can use custom mappers